Assignment 3/src/main/java/com/nuclei/assignment/service/graphoperations/GraphOperationsImpl.java

```diff
@@ -0,0 +1,227 @@
+package com.nuclei.assignment.service.graphoperations;
+
+import com.nuclei.assignment.constants.ExceptionsConstantsUtils;
+import com.nuclei.assignment.constants.SuccessConstantsUtils;
+import com.nuclei.assignment.entity.UserEntity;
+import com.nuclei.assignment.exception.CustomException;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.PriorityQueue;
+import java.util.Queue;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * Graph operations Implementation.
+ */
+public class GraphOperationsImpl implements GraphOperations {
+  /**
+   * The Users.
+   */
+  private final List<UserEntity> users;
+  
+  /**
+   * The Logger.
+   */
+  private final Log logger = LogFactory.getLog(GraphOperationsImpl.class);
+  
+  /**
+   * Graph operations Constructor.
+   */
+  public GraphOperationsImpl() {
+    users = new ArrayList<>();
+  }
+  
+  private Integer getIndexInList(final int id) {
+    return Collections.binarySearch(users, new UserEntity(id),
+      Comparator.comparing(UserEntity::getId));
+  }
+  
+  @Override
+  public List<UserEntity> getAllUsers() {
+    return users;
+  }
+  
+  @Override
+  public UserEntity getUserById(final int id) throws CustomException {
+    final int index = getIndexInList(id);
+    if (index < 0) {
+      logger.error(String.format(ExceptionsConstantsUtils.INVALID_ID, id));
+      throw new CustomException(String.format(ExceptionsConstantsUtils.INVALID_ID,
+          id));
+    }
+    return users.get(index);
+  }
+  
+  @Override
+  public boolean checkUserExistById(final int id) {
+    final int index = getIndexInList(id);
+    return index >= 0;
+  }
+  
+  @Override
+  public void createUser(final UserEntity user) {
+    final int index = getIndexInList(user.getId());
```

Comment:
should we use a list or a map for the graph?

Reply:
Done with Map